components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -682,6 +684,96 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
          endif
 
          nEdges = nEdgesArray(3)
+         if (config_GM_horizontal_taper == 'RossbyRadius'.or. &
+             config_Redi_horizontal_taper == 'RossbyRadius') then
+            allocate (RossbyRadiusTaper(nEdges))
+            call mpas_pool_get_array(meshPool, 'fEdge', fEdge)
+
+            !$omp parallel
+            !$omp do schedule(runtime) private(k, cell1, cell2, sum_hN, lt1, lt2, c_mode1, resolution, dc_Lr)
+            do iEdge = 1, nEdges
+               cell1 = cellsOnEdge(1, iEdge)
+               cell2 = cellsOnEdge(2, iEdge)
+               sum_hN = 0.0
+
+               do k = minLevelEdgeBot(iEdge)+1, maxLevelEdgeTop(iEdge)-1
+
+                  lt1 = 0.5_RKIND*(layerThickness(k,cell1) + layerThickness(k-1,cell1))
+                  lt2 = 0.5_RKIND*(layerThickness(k,cell2) + layerThickness(k-1,cell2))
+                  sum_hN = sum_hN + 0.5_RKIND*(lt1*sqrt(max(BruntVaisalaFreqTop(k,cell1),0.0_RKIND)) + &
+                                  lt2*sqrt(max(BruntVaisalaFreqTop(k,cell2),0.0_RKIND)))
+               end do
+
+               ! See Chelton (1998)
+               ! https://journals.ametsoc.org/view/journals/phoc/28/3/1520-0485_1998_028_0433_gvotfb_2.0.co_2.xml
+               ! eqn 2.2a, also Appendix A,
+               ! for the first-mode (m=1) baroclinic gravity wave speed in m/s
+               ! c_m = 1/pi integral(N dz )
+
+               c_mode1 = max(1.0E-5_RKIND, sum_hN/3.141592_RKIND)
+               cGMphaseSpeed(iEdge) = c_mode1
+
+               ! See Hallberg (2013) https://doi.org/10.1016/j.ocemod.2013.08.007
+               ! just after eqn 4 the defines the deformation
+               ! radius as:
+               ! L = sqrt( c_g^2/ ( f^2 + 2*beta*c_g) )
+               !   = c_g * sqrt(1/( f^2 + 2*beta*c_g) )  (note c_g is guaranteed non-negative)
+               ! This comes from a combination of Chelton (1998) eqns 2.3a and
+               ! 2.3b, with a smooth transition in between.
+               RossbyRadius(iEdge) = c_mode1*sqrt( 1.0_RKIND / &
+                  (fEdge(iEdge)*fEdge(iEdge) + 2.0_RKIND*betaEdge(iEdge)*c_mode1 + 1.0E-15_RKIND))
+
+               dc_Lr = dcEdge(iEdge)/RossbyRadius(iEdge)
+
+               ! Ramp the parameterization between the given min and max values
+               ! of the ratio of the cell width to the Rossby radius
```

Comment:
Following my discussions with @xylar and @vanroekel on slack, we agreed that the check on `c_mode1` in the if statement below is not appropriate. If we really wanted to include it, then it should lead to a full-on GM (`RossbyRadiusTaper=1`). So I tried two tests: 1) I removed the check completely, and 2) I changed the first part of the if statement as follows:
```
if (dc_Lr >= config_GM_Rossby_ramp_max .or. c_mode1.lt.0.1_RKIND) then
       RossbyRadiusTaper(iEdge) = 1.0_RKIND
else if (dc_Lr <= config_GM_Rossby_ramp_min ) then
       RossbyRadiusTaper(iEdge) = 0.0_RKIND 
```

In both cases, the ocean died at the very first time step because of nans in the state variables. I don't understand why this happens, since this only involves a check modification.

Reply:
@milenaveneziani could you let me know what your redi settings are for your tests?  Also is this E&G GM?

Reply:
Thanks Luke. I am using default Redi and GM Visbeck for these.

Reply:
These are B-cases right?  So constant 400 for Redi?  I was thinking perhaps the instability was in Redi if you were using equal GM

Reply:
yes, B-cases. and yes, Redi is constant at 400. We just wanted to test the GM changes for this one.

Reply:
Thanks for confirming that.  One other question, just so I understand, does the code run stably if it is like on this PR, where the c_mode1 check is in the wrong place?  And if you remove that incorrect check it crashes?

Reply:
Also does the model run without the rossby radius taper option on?

Reply:
exactly. I was able to run fine for 20 years with the new Rossby Radius ramp represented by the if statement with the c_mode1 check in the wrong place.